lib/facility.rb

```diff
@@ -1,14 +1,34 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_details)
+    @name = facility_details[:name]
+    @address = facility_details[:address]
+    @phone = facility_details[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = facility_details[:collected_fees]
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @services.include?('Vehicle Registration')
+      vehicle.set_plate_type
+      vehicle.collect_fees
```

Comment:
This method does not exist on a vehicle - it exists in this class so I would expect something like `collect_fees(vehicle)`